File: mlpack/lars.cpp
#include <mlpack/core.hpp>
#include "lars.hpp"

using namespace mlpack;
using namespace mlpack::regression;

int main(int argc, char** argv)
{
    arma::mat X, Y;
    
    // Load the data.
    data::Load("../data/artificial_data_X.csv", X, true, false);
    data::Load("../data/artificial_data_Y.csv", Y, true, false);
    
    // Perform LARS.
    LARS lars(0, 0, false);
    arma::vec beta;
    lars.Regress(X, Y.unsafe_col(0), beta, false);
    
    // Show the results.
    std::cout << beta << std::endl;
    
    return 0;
}

File: mlpy/lars.py
import numpy as np
import os
import mlpy
from Lars import LARS

# Load the data.
f = open(os.path.dirname(__file__) + '../data/artificial_data_X.data')
X = np.fromfile(f, dtype=np.float64, sep=' ')
X = X.reshape(-1, 5)
f.close()

f = open(os.path.dirname(__file__) + '../data/artificial_data_Y.data')
Y = np.fromfile(f, dtype=np.float64, sep=' ')
f.close()

# Perform LARS.
lars = LARS()
lars.learn(X, Y)

# Show the results.
print lars.beta()